samples/build/build_source-to-image_cr.yaml

```diff
@@ -6,8 +7,8 @@ spec:
   source:
     url: https://github.com/sclorg/nodejs-ex
   strategy:
-    name: "source-to-image"
-    namespace: "openshift"
-  builderImage: "docker.io/centos/nodejs-10-centos7"
+    name: source-to-image
+    namespace: openshift
+  builderImage: registry.redhat.io/rhscl/nodejs-12-rhel7:latest
```

Comment:
Curious, does this work without auth?

Reply:
You're correct, it will require user authentication. And, considering the scenario where it's provided `.spec.output.credentials.name` it should also cover any authentication needed during image-pulll as well.

Should we then consider a different construction? For instance:

```yml
spec:
[...]
  # introduce images to handle all container image references in a single place
  images:
    # credentials become a list of dockercfg typed secrets
    credentials:
      - docker-registry-secret
      - quayio-registry-secret
    # builder image, former spec.builderImage
    builder: docker.io/builder/image:latest
    # output image, kept in place
    output: quay.io/output/image:latest
```

Reply:
Thanks for exposing this use case!
I would rather prefer keeping things contextual at the top-level : 
https://github.com/redhat-developer/build/pull/38